File: typescript-analysis/src/repositories/OrderRepository.ts
import { Repository } from 'typeorm';
import { AppDataSource } from '../config/database';
import { Order } from '../entities/Order';

export class OrderRepository {
  private repository: Repository<Order>;

  constructor() {
    this.repository = AppDataSource.getRepository(Order);
  }

  async findById(id: string): Promise<Order | null> {
    return await this.repository.findOne({ where: { id } });
  }

  async findAll(): Promise<Order[]> {
    return await this.repository.find();
  }

  async save(entity: Order): Promise<Order> {
    return await this.repository.save(entity);
  }

  async delete(id: string): Promise<void> {
    await this.repository.delete(id);
  }
}

File: extracted-python/repositories/OrderRepository.py
# Repository pattern implementation for Order
from sqlalchemy.orm import Session
from typing import List, Optional
from models.order import Order

class OrderRepository:
    def __init__(self, db: Session):
        self.db = db

    def find_all(self) -> List[Order]:
        return self.db.query(Order).all()

    def find_by_id(self, id: int) -> Optional[Order]:
        return self.db.query(Order).filter(Order.id == id).first()

    def save(self, entity: Order) -> Order:
        self.db.add(entity)
        self.db.commit()
        self.db.refresh(entity)
        return entity

    def delete_by_id(self, id: int) -> bool:
        entity = self.find_by_id(id)
        if entity:
            self.db.delete(entity)
            self.db.commit()
            return True
        return False
